Update `cadence`/`flow-go`/`flow-go-sdk`/`flow-emulator` dependencies

m-Peter opened the pull request:
## Description



______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-nft/blob/master/CONTRIBUTING.md#styleguides).
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Dependency Updates**
	- Upgraded Go version from 1.22 to 1.23
	- Updated multiple dependencies across Flow project modules
	- Updated testing libraries and OpenTelemetry instrumentation packages
	- Added new dependency: `github.com/onflow/bridged-usdc/lib/go/contracts` at version 1.0.0

- **Infrastructure**
	- Updated base image in Dockerfile to Go version 1.23
	- Adjusted CI configuration to use Go version 1.23 for testing and linting

- **Bug Fixes**
	- Refined expected outcomes in transaction tracing tests to align with updated logic
	- Removed unnecessary logging of transaction execution details, streamlining processing
	- Adjusted expected values in various test cases to reflect changes in underlying logic
	- Corrected expected block numbers in transaction filter tests to match actual outputs
	- Updated expected results for contract call assertions to reflect new logic for flow block height
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

m-Peter commented on .github/workflows/ci.yml:
```diff
@@ -16,7 +16,7 @@ jobs:
     - uses: actions/setup-go@v4
       with:
-        go-version: '1.22.x'
+        go-version: '1.23.x'
```
Well, this says otherwise: https://go.dev/blog/go1.23

m-Peter commented on .github/workflows/ci.yml:
```diff
@@ -33,7 +33,7 @@ jobs:
       - uses: actions/setup-go@v4
         with:
-          go-version: "1.22.x"
+          go-version: "1.23.x"
```
Well, this says otherwise: https://go.dev/blog/go1.23

m-Peter commented on bootstrap/bootstrap.go:
```diff
@@ -364,10 +363,6 @@ func (b *Bootstrap) StartMetricsServer(ctx context.Context) error {
 	b.logger.Info().Msg("bootstrap starting metrics server")
 
 	b.metrics = flowMetrics.NewServer(b.logger, uint(b.config.MetricsPort))
-	err := util.WaitClosed(ctx, b.metrics.Ready())
```
That's a good question! So the metrics server became a component in this commit: https://github.com/onflow/flow-go/commit/4c8ac17f20797161d691d390f0598098ad47fce5.

Which means that with:
```go
err := util.WaitClosed(ctx, b.metrics.Ready())
```

The entire test suite would fail, due to indefinite blocking. The same goes for the shutdown of the metrics server:
```go
<-b.metrics.Done()
```
I had to remove the above line.

The flow-go commit I linked above, was intended to be used from this PR: https://github.com/onflow/flow-evm-gateway/pull/682. I am unsure as to how to proceed here..

janezpodhostnik replied:
I'll take a look today.

janezpodhostnik replied:
This should do it I think: https://github.com/onflow/flow-evm-gateway/pull/741

m-Peter replied:
> This should do it I think: #741

@janezpodhostnik That piece of code works like a charm :100: Could you prepare that PR and merge on this one?

janezpodhostnik replied:
merged!

m-Peter commented:
@turbolent I have updated this PR to use the latest tagged release of `flow-emulator`. And while at it, fixed some breaking changes which occurred in the integration tests.

peterargue commented on bootstrap/bootstrap.go:
```diff
@@ -368,20 +368,37 @@ func (b *Bootstrap) StartMetricsServer(ctx context.Context) error {
+
+	ictx, errCh := irrecoverable.WithSignaler(ctx)
+	b.metrics.Start(ictx)
+	<-b.metrics.Ready()
```
if the context is cancelled before the metric server is running, this will block forever, 
```suggestion
	if err := util.WaitClosed(ctx, b.metrics.Ready()); err != nil {
		return fmt.Errorf("failed to start metrics server: %w", err)
	}
```

peterargue reviewed (approved):
added a comment about a corner case for the metrics server, but otherwise looks good
